### DISCUSSION TITLE
Switch between Dark and Light mode

### DISCUSSION THREAD

**Discussion by jasonvriends** [tabler/tabler#1365] [Q&A]
I would like to save the theme as part of a user profile. Is it possible to change from Dark and Light modes without a query string i.e. ?theme=dark

**Comment by WellThisIsUsed** [comment]
Forgive me I don't quite understand the question.
But I'll try and help, anyways.

Dark Mode and Light Mode are controlled by the class on the body. (ie: `<body class="theme-light">` and `<body class="theme-dark">`).

Personally for each user, I have a row in a `Preferences` table in the database which stores their preference.
With the default being Light Mode.

And, when rendering the HTML I just need to use `<body class="${this.Request.Preferences.DarkMode ? 'theme-dark' : 'theme-light'}">`.

There are other ways of implementing this depending on how your app is structed, and what frameworks you're using.

A database is an option, but so is storing this preference as a Cookie or in the Local Storage of the users browser (how the [Tabler preview](https://preview.tabler.io/) does it).

**Reply by jasonvriends** [reply]
This is exactly what I was after, thank you! I had to remove ```<script src="{% static 'js/demo-theme.min.js' %}" defer></script>``` as that was enforcing ```theme-light``` on the body no matter what. I am using Django and trying to create a starter project for anyone starting a new project. https://github.com/jasonvriends/django-starter
